src/ZfrCors/Factory/CorsServiceFactory.php

```diff
@@ -35,11 +36,20 @@ class CorsServiceFactory implements FactoryInterface
      * {@inheritDoc}
      * @return CorsService
      */
-    public function createService(ServiceLocatorInterface $serviceLocator)
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
     {
         /* @var $corsOptions CorsOptions */
-        $corsOptions = $serviceLocator->get('ZfrCors\Options\CorsOptions');
+        $corsOptions = $container->get('ZfrCors\Options\CorsOptions');
 
         return new CorsService($corsOptions);
     }
+    /**
+     * {@inheritDoc}
+     *
+     * @return CorsService
+     */
+    public function createService(ServiceLocatorInterface $container)
```

Comment:
> Using __invoke makes this class callable, same as a closure or anonymous function so yes it will work.

@macnibblet as I said I tested your suggestion by removing `FactoryInterface` and `createService()` and tests failed. So if you think that I should change something please explain more about what and why should be changed.